### PR TITLE
Change target names

### PR DESCRIPTION
**Pull request by tizianoGuadagnino**
In the current setting, if someone will `FetchContent` KISS, there might be conflicts based on target names, as `core` and `pipeline` are very common. There is probably another workaround to this, but it makes more sense to remove the aliasing and call every target `kiss_icp_<something>`. This is the safest option that should not fire back later on, I attach here an example of what I am talking about.  

image

### CODE REVIEW

**Comment by tizianoGuadagnino** [comment]
#328 introduce the change in the build system that will also fix FetchContent

**Review by benemer — APPROVED** [review]

**Comment by nachovizzo** [comment]
I knew this might happen sooner or later. I think the underlying problem is not the name of the targets but the lack of an installable C++ library. 

I postponed this always, but I'd say we could now fix this the proper way. Using modern cmake, and relying on exported targets instead of manually adding directories in the current build space

A probably outdated example:
https://github.com/PRBonn/vdbfusion/blob/main/src/vdbfusion/vdbfusion/cmake/Install.cmake

 In summary,  I think we should make the  C++ library installable, like is the case with all the dependencies we use

**Comment by tizianoGuadagnino** [comment]
Absolutely agree with @nachovizzo, that should be the proper way. Shall we merge this and open a new branch for the _install_ ?

**Comment by nachovizzo** [comment]
> Absolutely agree with @nachovizzo, that should be the proper way. Shall we merge this and open a new branch for the _install_ ?

I'm fine with it as long as we commit to do it and review it relatively soon

I will approve this so I don't block you but please let's revisit this soon. I won't have time in the next weeks to work on this particular build feature

**Review by nachovizzo — APPROVED** [review]

**Comment by tizianoGuadagnino** [comment]
You got my word.
